src/controllers/recipe.controller.ts

```diff
@@ -97,11 +101,9 @@ export class RecipeController {
       }
 
       if (updatedRecipe.OK && updatedRecipe.recipe) {
-        webhookService.webhookHandler(
-          userId,
-          "update_recipe",
-          updatedRecipe.recipe
-        );
+        const jsonedRecipe = updatedRecipe.recipe?.toJSON();
+        jsonedRecipe._id = jsonedRecipe._id.toString();
```

Comment:
Is this necessary? Won't it be converted to the string after using `toJSON()`?

Reply:
@arkus7 I spent all day on this and _id is not converted to string. Btw userId (mongodb ID) is normal string even when not calling .toJSON().

Reply:
These changes were made for sinon to pass tests 
`      sinon.assert.calledWith(
        sendWebhookNotificationSpy,
        userId,
        "update_recipe",
        response.body
      );`

Was throwing error that response.body._id is different than updatedRecipe._id
![image](https://user-images.githubusercontent.com/59976377/128026062-32c62600-a1f3-4f52-8034-211563a144fb.png)

Red color is expected value, green is what was passed to webhookHandler() and blue are values that match

Reply:
@Abrafaks You can use `transform` option on schema's `toJSON` method

```ts
schema.set('toJSON', {
  transform: (doc: RecipeDocument, ret: unknwon, options: unknown) => {
    return { ...ret, _id: doc._id.toString() };
  }
});
```
This will return all properties of `ret` (which is the document converted to plain object), but will override `_id` with `document._id.toString()` which is what you do on line 105.
This would be then done wherever you use a `Document.toJSON()` method.
More info here: https://mongoosejs.com/docs/api.html#document_Document-toObject